data_steward/.gcloudignore

```diff
@@ -0,0 +1,98 @@
+# This file specifies files that are *not* uploaded to Google Cloud Platform
+# using gcloud. It follows the same syntax as .gitignore, with the addition of
+# "#!include" directives (which insert the entries of the given .gitignore-style
+# file at that point).
+#
+# For more information, run:
+#   $ gcloud topic gcloudignore
+#
+.gcloudignore
```

Comment:
There are multiple redundancies between `.gitignore` and this `.gcloudignore` file.  Why not make use of the include directive?
```
#!include:.gitignore
```
Then you can remove lines 13-33, 37-39, 42-53, 55-59, 63, 64, 68-73, 79-80.

Reply:
I like the idea of removing redundancy, but I avoided the include for fear of repeating a prior experience. I did something like this previously and thereby excluded a file from deployment unintentionally. The resulting error I encountered in prod `module 'xyz' not found` only occurred several hours later and was not easy to figure out at the time or to reproduce in tests (because gcloudignore affects the runtime environment only). I believe files like `{app,cron}.yaml` we **should** include in gitignore but not in gcloudignore. Does this make sense?

Reply:
I understand your hesitation.  But we are not currently ignoring `{app,cron}.yaml` files in our `.gitignore` file.  Even if we were, we can explicitly add those files to be included with the deployment by adding these:
```
!app.yaml
!cron.yaml
```
to the `.gcloudignore` file.  Also, if we need to list the files that will be uploaded we can use the `gcloud meta list-files-for-upload` command.

Out of curiousity, why should those files be excluded from git?  I've been on projects where configuration files were templated and version controlled to make sure we knew what variables had to be set and how.  Then each dev had a functional copy of the file.  To make sure we didn't store passwords/keys/secrets, we ran `git update-index --assume-unchanged <filepath>` to prevent us from accidentally uploading credentials.  https://stackoverflow.com/questions/10755655/git-ignore-tracked-files

If you prefer to err on the side of caution, I understand.  It just seems like both files are adding a lot of boilerplate to ignore packages/imports/files that we aren't even using in the project.

Reply:
Final decision is yours.  Please don't forget to update the file to include the line `*_env/`.

Reply:
Thanks @lrwb-aou! I didn't know about that meta command. This makes me more comfortable about including `.gitignore` in `.gcloudignore` so I will look into it. :)

Prior DRC discussions suggested we should be cautious about any secrets or identifiers for cloud resources in our repositories. I think we agree our approach has to strike a balance between confidentiality and ease of deployment- paranoia could mean terribly incovenient/error-prone deployment. Yes to templating the yaml files! I am [using jinja to template cron.yaml](https://github.com/all-of-us/curation/blob/develop/data_steward/resources/templates/cron.tpl.yaml) and would like for us to do something similar for `app.yaml`. Please **do** share any suggestions you may have on how engineers can better share app secrets.

Reply:
I'm in complete agreement about keeping secrets to ourselves.  We definitely don't want to version any of that.

I love the cron template.  I haven't personally used jinja often, but I know it's a powerful tool and the template is very clean and understandable.  I'll look around to see if there is anything better than telling git to ignore file changes.  I would think there is.